### ISSUE TITLE
Glitching while trying to use the editor

### ISSUE THREAD

**Issue by Darkteku** [goatfungus/NMSSaveEditor#969]
i have tried reinstalling Java 8 but still after like 3 seconds of using the editor it flickers all over the screen glitches like crazy has any one fixed this issue or know how to fix it?

**Comment by MENXIN-ZYT** [comment]
Me too, I have reinstalled java8 and there is no other version of java on my computer, but I have no idea why I still have this problem.
Maybe the source of the problem is the nvdia graphics driver?

**Comment by MENXIN-ZYT** [comment]
> i have tried reinstalling Java 8 but still after like 3 seconds of using the editor it flickers all over the screen glitches like crazy has any one fixed this issue or know how to fix it?

May I ask if you have solved this problem?
I've seen a lot of people experiencing this problem as well

**Comment by Darkteku** [comment]
> > i have tried reinstalling Java 8 but still after like 3 seconds of using the editor it flickers all over the screen glitches like crazy has any one fixed this issue or know how to fix it?
> 
> May I ask if you have solved this problem? I've seen a lot of people experiencing this problem as well

i have not actually not sure if its a windows 11 thing or not but i will say it used to work before the omega update on windows 11 not since omega and now orbital it still glitches after 3 seconds.

ive tried lowering my refresh rate. ive tried opening it in java binaries but no luck there maybe another code software i could try but since it already runs java i dont want to mess with it.

nvidia control panel is a no go since i cant force it to use the amd graphics built into my cpu since its a jar and not an exe

**Comment by curlymoe** [comment]
For me it is a Windows 11 problem, I dual boot Win 10 and 11, and I only have the problem on 11. Because I play the game on Win 11, to do any save editing I drop the save over to Win 10, edit it and then pass it back to win 11. A bit of a pain but at least it works.

**Comment by sgrad** [comment]
I had a similar problem, after hovering over any of the inventory items (exosuit, ship, multitool, etc) random bits of the main window would show over the main window.

Potential fix (worked for me):
- find your main java executable, either:
  - it's in `C:\Program Files\Java\jre-1.8\bin\javaw.exe`
  - or, open the `Java Control Panel` (use win search) and you can see the binary location under `Java` tab -> click `View` -> `Path` column of the 1.8 binary
- Open the location of `javaw.exe` in a standard `explorer` window and right click -> `Properties` -> `Compatibility` tab
  - enable Win 8 Compatibility
  - tick `Disable Full Screen`
  - [optional] tick the run as administrator (be careful if you run other untrusted java things)

Thanks to @Darkteku and @curlymoe for suggesting it might be a win 11 issue.

**Comment by goatfungus** [comment]
Thanks @sgrad for the info.

I might see if i can add that to the FAQ's since a lot of different people experience this.

**Comment by Darkteku** [comment]
> I had a similar problem, after hovering over any of the inventory items (exosuit, ship, multitool, etc) random bits of the main window would show over the main window.
> 
> TL;DR run the `javaw.exe` binary in win 8 compatibility
> 
> Potential fix (worked for me):
> 
> * find your main java executable, either:
>   
>   * it's in `C:\Program Files\Java\jre-1.8\bin\javaw.exe`
>   * or, open the `Java Control Panel` (use win search) and you can see the binary location under `Java` tab -> click `View` -> `Path` column of the 1.8 binary
> * Open the location of `javaw.exe` in a standard `explorer` window and right click -> `Properties` -> `Compatibility` tab
>   
>   * enable Win 8 Compatibility
>   * tick `Disable Full Screen`
>   * [optional] tick the run as administrator (be careful if you run other untrusted java things)
> 
> Thanks to @Darkteku and @curlymoe for suggesting hinting at a win 11 issue.

sadley that did not work for me at all took 4 seconds instead of 3 before it started glitching out

**Comment by sgrad** [comment]
A couple of things I can think of:
 - Double check that you have set the compatibility options for the correct binary, it's `javaW.exe` (*not* `javaWS.exe`)
 - Try using the `Change settings for all users` button to apply the compatibility fixes
 - Reboot
 - Make sure you have all the windows updates
 - Perform a clean graphics driver reinstall - for nvidia it's an option when running the installer, for amd I'm not sure, sorry

Hope this helps :)

**Comment by curlymoe** [comment]
> A couple of things I can think of:
> 
>     * Double check that you have set the compatibility options for the correct binary, it's `javaW.exe` (_not_ `javaWS.exe`)
> 
>     * Try using the `Change settings for all users` button to apply the compatibility fixes
> 
>     * Reboot
> 
>     * Make sure you have all the windows updates
> 
>     * Perform a clean graphics driver reinstall - for nvidia it's an option when running the installer, for amd I'm not sure, sorry
> 
> 
> Hope this helps :)

Thanks for all your efforts with this, but I'm afraid it did not work for me, still just the same. At least I can drop back into win 10 to use the Editor.

**Comment by Darkteku** [comment]
apparently the issue has been fixed with the most recent update not sure what happened

**Comment by curlymoe** [comment]
> apparently the issue has been fixed with the most recent update not sure what happened

Wish it had for me, but still the same.

**Comment by BingoFuel5X5** [comment]
> i have tried reinstalling Java 8 but still after like 3 seconds of using the editor it flickers all over the screen glitches like crazy has any one fixed this issue or know how to fix it?

Mine has done that with the past few versions.  Not a big deal, though.  If you max/min the window, it straightens out.

**Comment by BobrDobr27** [comment]
> > i have tried reinstalling Java 8 but still after like 3 seconds of using the editor it flickers all over the screen glitches like crazy has any one fixed this issue or know how to fix it?
> 
> Mine has done that with the past few versions. Not a big deal, though. If you max/min the window, it straightens out.

It's working! THX
Yes, on Win 10 all was good, this trouble with Win 11

**Comment by SirCathal** [comment]
For me the Option -Dsun.java2d.d3d=false does the trick.
My complete Start Batch is now:

```
@echo off
:start
java --enable-native-access=ALL-UNNAMED -Xmx4096M -Dsun.java2d.d3d=false -jar NMSSaveEditor.jar -autoupdate
if errorLevel 2 (
	move /Y ~NMSSaveEditor.dl NMSSaveEditor.jar
	goto start
)
if errorlevel 1 (
	pause
)
```

-Dsun.java2d.d3d=false Disables Direct3D for 2D Painting. 
-Xmx4096M Sets the Object Limit to 4GB
--enable-native-access=ALL-UNNAMED heps to use OpenJDK with the Editor

With those settings the editor runs fine with Windows 11 25H2 patch level 08.2026
